regparser/tree/reg_text.py

```diff
@@ -83,6 +83,27 @@ def build_subpart(text, part):
     return struct.Node(
         "", [], label, subpart_title, node_type=struct.Node.SUBPART)
 
+def build_subjgrp(text, part):
+    """
+    We're constructing a fake "letter" here by taking the first letter of
+    each word in the subjgrp's title, or using the first two letters of the
+    first word if there's just one—we're avoiding single letters to make sure we
+    don't duplicate an existing subpart, and we're hoping that the initialisms
+    created by this method are unique for this regulation.
+    We can make this more robust by accepting a list of existing initialisms and
+    returning both that list and the Node, and checking against the list as we
+    construct them.
+    """
+    def subjgrp_label(candidate_text):
```

Comment:
Fair enough. One note on checking existing titles (this bit CFPB's HMDA api): when new data is added, in this case new titles, it may re-order your existing titles, breaking the node's lineage. That's a broader problem, however, which we also run in to when markerless paragraphs get re-ordered